src/parsers/trades/liquid/index.ts

```diff
@@ -0,0 +1,175 @@
+import { getCSVData } from '../../';
+import { EXCHANGES, IImport, IPartialTrade, ITrade } from '../../../types';
+import { createDateAsUTC, createID } from '../../utils';
+
+enum LiquidOrderDirection {
+    PAY = 'PAY',
+    RECEIVE = 'RECEIVE',
+}
+
+interface ILiquid {
+    exchange: string;
+    currency_type: string;
+    direction: string;
+    transaction_id: string;
+    transaction_type: string;
+    gross_amount: string;
+    currency: string;
+    execution_id: string;
+    generated_for_type: string;
+    generated_for_id: string;
+    transaction_hash: string;
+    from_address: string;
+    to_address: string;
+    state: string;
+    created_at_jst: string;
+    updated_at_jst: string;
+    created_at_utc: string;
+    updated_at: string;
+    notes: string;
+    used_id: string;
+    account_id: string;
+}
+
+function groupByExecutionID(group: any, line: any) {
+    group[line.execution_id] = group[line.execution_id] ?? [];
+    group[line.execution_id].push(line);
+    return group;
+}
+
+export default async function processData(importDetails: IImport): Promise<ITrade[]> {
+    const data: ILiquid[] = await getCSVData(importDetails.data) as ILiquid[];
+    const internalFormat: ITrade[] = [];
+    const sorted = data.reduce(groupByExecutionID, {});
+    for (const execution of Object.keys(sorted)) {
```

Comment:
I beliieve we can do a for in loop here instead which will expose the key and the value rather then doing for of on the keys

Reply:
I replaced with in loop